### ISSUE TITLE
Loader memory leak in setupLoaderTermPhysDevs function

### ISSUE THREAD

**Issue by scyganINTEL** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1449]
There seems to be a memory leak around 
https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/55867dbad6ae423b3bd78c15f6771031a710b5ad/loader/loader.c#L4997

```
new_phys_devs = loader_instance_heap_alloc(inst, sizeof(struct loader_physical_device_term *) * inst->total_gpu_count,
                                               VK_SYSTEM_ALLOCATION_SCOPE_INSTANCE);
```

When the next alloc after this line fails, the memory pointer by `new_phys_devs` is never freed.

This fails a VulkanCTS test `dEQP-VK.wsi.win32.swapchain.simulate_oom.*` from  `vulkan-cts-1.0.1` branch. 

I have not reproduced this on later CTS branches, as checks from this test were moved to some another test (?).

**Comment by scyganINTEL** [comment]
This seems to be a regression from `sdk-1.0.37`

**Comment by karl-lunarg** [comment]
Probably introduced in 8e9049915cf30da70579a58a9b8cd18a6b56d186

**Comment by MarkY-LunarG** [comment]
I've got a fix.  I'll test it through our automated system and push it up.  Likely on Monday.

**Comment by scyganINTEL** [comment]
Thanks!

**Comment by MarkY-LunarG** [comment]
Fixed with commit 56bf9519fbfe9b82166b29c9796b226160661889
